File: EVSDKDemo/EVSDKDemo/EVDemoRootViewController.m
//
//  EVDemoRootViewController.m
//  EVSDKDemo
//
//  Created by mashuaiwei on 16/7/25.
//  Copyright © 2016年 cloudfocous. All rights reserved.
//

#import "EVDemoRootViewController.h"
#import "EVSDKManager.h"
#import <CoreFoundation/CoreFoundation.h>
#import <Foundation/Foundation.h>
#import "CCAlertManager.h"

@interface EVDemoRootViewController ()
@property (weak, nonatomic) IBOutlet UITextField *appidField;
@property (weak, nonatomic) IBOutlet UITextField *appkeyField;
@property (weak, nonatomic) IBOutlet UITextField *appSecreteField;
@property (weak, nonatomic) IBOutlet UITextField *useridField;
@property (weak, nonatomic) IBOutlet UILabel *SDKVersionLabel;

@end

@implementation EVDemoRootViewController


#pragma mark - life circle

- (void)viewDidLoad{
    [super viewDidLoad];
    
    [self addObserver];
    self.SDKVersionLabel.text = [NSString stringWithFormat:@"SDK Version:%@", [EVSDKManager SDKVersion]];
}

- (void)dealloc{
    [[NSNotificationCenter defaultCenter] removeObserver:self];
}


#pragma mark - over write

- (BOOL)shouldPerformSegueWithIdentifier:(NSString *)identifier sender:(id)sender{
    if ([EVSDKManager isSDKInitedSuccess]) {
        return YES;
    }
    
    UIAlertView *alert = [[UIAlertView alloc] initWithTitle:nil message:@"SDK尚未初始化！" delegate:nil cancelButtonTitle:@"确定" otherButtonTitles:nil, nil];
    [alert show];
    return NO;
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    for (UIView *subviews in self.view.subviews) {
        [subviews endEditing:YES];
    }
}


#pragma mark - UIAction

- (IBAction)initSDKButtonClicked:(UIButton *)sender {
    [self initSDK];
}


#pragma mark - notifications

- (void)initSDKError:(NSNotification *)notification{
    NSLog(@"---notification:%@", notification);
    NSString *message = [NSString stringWithFormat:@"%@", notification.object];
    [[CCAlertManager shareInstance] performComfirmTitle:notification.name message:message comfirmTitle:@"确定" WithComfirm:nil];
}

- (void)initSDKSuccess{
    [[CCAlertManager shareInstance] performComfirmTitle:@"初始化SDK成功" message:nil comfirmTitle:@"确定" WithComfirm:nil];
}


#pragma mark - private methods

- (void)addObserver{
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(initSDKError:) name:EVSDKInitErrorNotification object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(initSDKSuccess) name:EVSDKInitSuccessNotification object:nil];
}

- (void)initSDK{
    NSString *appid = self.appidField.text;
    NSString *appkey = self.appkeyField.text;
    NSString *appsecret = self.appSecreteField.text;
    NSString *userid = self.useridField.text;
    
    // 一般
    [EVSDKManager initSDKWithAppID:appid appKey:appkey appSecret:appsecret userID:userid];
}

@end

File: EVSDKDemo_Swift/EVDemoRootViewController.swift
//
//  EVDemoRootViewController.swift
//  EVSDKDemo
//
//  Created by Lcrnice on 17/5/19.
//  Copyright © 2017年 cloudfocous. All rights reserved.
//

import EVSDKBaseFramework
import EVMessageFramework

class EVDemoRootViewController: UIViewController {
  @IBOutlet weak var appIDTF: UITextField!
  @IBOutlet weak var appKeyTF: UITextField!
  @IBOutlet weak var appSecreteTF: UITextField!
  @IBOutlet weak var userIDTF: UITextField!
  @IBOutlet weak var SDKVersionLabel: UILabel!
  
  deinit {
    NotificationCenter.default.removeObserver(self)
  }
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    self.addEVObserver()
    SDKVersionLabel.text = "SDK Version:\(EVSDKManager.sdkVersion())"
  }
  
  override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
    view.endEditing(true)
  }
  
  override func shouldPerformSegue(withIdentifier identifier: String, sender: Any?) -> Bool {
    if EVSDKManager.isSDKInitedSuccess() {
      return true
    }
    
    CCAlertManager.shareInstance().performComfirmTitle(nil, message: "SDK尚未初始化！", comfirmTitle: "OK", withComfirm: nil)
    return false
  }
  
  @IBAction func initSDKAction(_ sender: Any) {
    EVSDKManager.initSDK(withAppID: appIDTF.text!, appKey: appKeyTF.text!, appSecret: appSecreteTF.text!, userID: userIDTF.text!)
  }
}

// MARK: - 注册SDK
extension EVDemoRootViewController {
  fileprivate func addEVObserver() {
    NotificationCenter.default.addObserver(forName: .EVSDKInitSuccess, object: nil, queue: OperationQueue.main, using: { _ in
      CCAlertManager.shareInstance().performComfirmTitle("初始化成功", message: nil, comfirmTitle: "OK", withComfirm: nil)
      print("SDK 初始化成功")
    })
    
    NotificationCenter.default.addObserver(forName: .EVSDKInitError, object: nil, queue: OperationQueue.main, using: { notify in
      print("SDK 初始化失败")
      CCAlertManager.shareInstance().performComfirmTitle(notify.name.rawValue, message: String(format: "%@", notify.object as! CVarArg), comfirmTitle: "ok", withComfirm: nil)
    })
  }
}
